src/main/java/seedu/address/storage/GemCommandStorage.java

```diff
@@ -157,12 +158,15 @@ public void setGetModules(String path) throws IOException, IllegalValueException
      * @param modules List of Modules.
      * @return String of Module Codes and Modular Credits.
      */
-    public StringBuilder moduleExtractor(ObservableList<Module> modules) {
+    public StringBuilder moduleExtractor(ObservableList<Module> modules, Model model) {
```

Comment:
Actually, now that this method has to interact with model and deal with some filtering logic, should it be in the GemCommand class instead?

Reply:
Agreed - idk how I missed this out the first time round paiseh >< Since it's in the `Storage` component, it might be good if it's only concerned about storing and retrieving data without being aware of the model

Reply:
good point! will make the amendments right away 😸